pandas/core/indexes/base.py

```diff
@@ -3018,7 +3020,12 @@ def _convert_listlike_indexer(self, keyarr, kind=None):
 
     @Appender(_index_shared_docs["_convert_arr_indexer"])
     def _convert_arr_indexer(self, keyarr):
```

Comment:
would it be simpler to override DatetimeIndex._convert_arr_indexer and PeriodIndex._convert_arr_indexer?